460 Use HDF5 headers for file info

pford opened the pull request:
Closes #460 with FITS fix in PR #827.

Uses Hdf5Attributes to get header strings (via CartaHdf5Image) and set header_entries in FileInfoExtended.  This fixes shortcomings in casacore ImageHeaderToFITS including (1) removing headers with non-FITS keywords (such as HDF5 schema and converter headers), (2) adding FITS keywords (BLANK), and (3) sometimes computing headers which do not exist in image headers as described in #460.

pford committed:
Read HDF5 attributes for file info headers

pford committed:
Convert HDF5 header strings to header entries in extended file info

pford committed:
Remove unused include

kswang1029 reviewed (approved):
as far as I can tell with some hdf5 images I have, this PR works fine. 👍

confluence commented:
When I run this branch (with the latest frontend) the backend crashes after logging `[critical] stod` when I attempt to display any HDF5 file in the file browser.

kswang1029 commented:
> When I run this branch (with the latest frontend) the backend crashes after logging `[critical] stod` when I attempt to display any HDF5 file in the file browser.

Could you send me a small hdf5 image to test with?

confluence commented:
Hmm, it looks like it's not *any* file -- I'm checking in more detail which files trigger the issue and which do not.

confluence commented:
I have identified two images that are affected. They are both very large, but I'm attaching the headers here (since this is a header parsing issue).

I think that I know what the issue is with `XMMLSS_3_contcube` -- it has a `RESTFRQ` value of `4.940656458412E-324`, which is too small to be representable by a double, and this is causing `stod` to fail. I had to [add  a workaround to the HDF5 converter](https://github.com/idia-astro/fits2idia/commit/a3d7343d46881cfefab9007d6caeaf05125da803) to deal with this by parsing the value as a long double and converting to double (I don't recall whether we previously also had to handle this case in any of the CARTA header parsing code).

I can't locate a similar small value in the `ultravista` header, so that seems to be a different string to double parsing issue.

* [ultravista-dr3-k-band.fits.header.txt](https://github.com/CARTAvis/carta-backend/files/6693401/ultravista-dr3-k-band.fits.header.txt)
* [XMMLSS_3_contcube.fits.header.txt](https://github.com/CARTAvis/carta-backend/files/6693402/XMMLSS_3_contcube.fits.header.txt)

pford committed:
Fix HDF5 header numeric entries

Do not try to convert quoted values.  If stoi or stod out of range,
use long conversion.

pford commented:
Pushed a fix for the headers:  if the value starts with a single quote, do not convert to int/double.  If conversion to int/double throws out_of_range exception, try to convert to long int/double.

confluence commented:
I can confirm that this fixes the issue with the two files I tested. Every HDF5 file I tried now works.

kswang1029 commented:
Re-tested with 6dad03e and it still works fine with my few test hdf5 images. Ready to merge. 👍
